python/tvm/meta_schedule/database/json_file.py

```diff
@@ -0,0 +1,37 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""The default database that uses a JSON File to store tuning records"""
+from tvm._ffi import register_object
+
+from .. import _ffi_api
+from .database import Database
+
+
+@register_object("meta_schedule.JSONFile")
+class JSONFile(Database):
+    def __init__(
+        self,
+        record_path: str,
+        workload_path: str,
+        allow_missing: bool = True,
+    ) -> None:
+        self.__init_handle_by_constructor__(
+            _ffi_api.DatabaseJSONFile,  # type: ignore # pylint: disable=no-member
+            record_path,
+            workload_path,
+            allow_missing,
+        )
```

Comment:
I'm not sure I grok the design of working directly with a JSON wrapper like this instead of with a Database?

Reply:
I'm not sure I understand the question. This design follows the convention of AutoTVM and AutoScheduler where each line of a JSON file is a tuning record. Given the design is unlikeable in production, we allow developers to inherit from `PyDatabase` to interface with their own database, and this isn't limited to using JSON - they can actually use any format they like to serialize these records, as long as they override the 4 APIs required in `PyDatabase`